src/models/appointment.ts

```diff
@@ -0,0 +1,44 @@
+import mongoose from "mongoose";
+
+export interface IAppointment {
+    patient: string;
+    doctor: string;
+    date: Date;
+    time: string;
+    type: string;
+    link: string;
+}
+
+export interface IAppointmentModel extends IAppointment, mongoose.Document { }
+
+const appointmentSchema = new mongoose.Schema({
+    patient: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+        required: true,
+    },
+    doctor: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+        required: true,
+    },
+    date: {
+        type: Date,
+        required: true,
+    },
+    time: {
```

Comment:
I think we could use ISO date format to store in a single column the date and the time for an appointment

Reply:
I thought it will be easier approach for front-end part ( so they dont have to split it if they want to). We can use ISO date format too.

Reply:
The front end will import parsers for this functionality. It's simpler for the db with a single column